File: Python/decorator_pattern/cafe_decorator.py
from cafe import Cafe

class CafeDecorator(Cafe):
    def __init__(self, cafe):
        self.cafe = cafe
        
    def get_descricao(self):
        return self.cafe.get_descricao()
    
    def get_custo(self):
        return self.cafe.get_custo()

File: Java/com.faculdadevincit.designpatterns.decoratorpattern/src/CafeDecorator.java
public abstract class CafeDecorator implements ICafe {
    protected ICafe cafe;

    public CafeDecorator(ICafe cafe) {
        this.cafe = cafe;
    }

    @Override
    public double getCusto() {
        return cafe.getCusto();
    }

    @Override
    public String getDescricao() {
        return cafe.getDescricao();
    }

}
